### PR TITLE
Hide profile error when non-vets cannot load service history

### PR DESCRIPTION
**Pull request by erikphansen**
## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile/components/ProfileWrapper.jsx [hunk 1]
```diff
@@ -107,7 +107,7 @@ const ProfileWrapper = ({
 const mapStateToProps = (state, ownProps) => {
   const veteranStatus = selectProfile(state)?.veteranStatus;
   const invalidVeteranStatus =
-    !veteranStatus || veteranStatus === 'NOT_AUTHORIZED';
+    !veteranStatus || veteranStatus.status === 'NOT_AUTHORIZED';
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/ProfileWrapper.jsx:110]: this was the root of the problem 🤦

File: src/applications/personalization/profile/tests/components/alerts/Profile.data-unavailable-error.unit.spec.js [hunk 2]
```diff
@@ -144,16 +140,40 @@ describe('Profile "Not all data available" error', () => {
     expect(view.queryByTestId(ALERT_ID)).not.to.exist;
   });
 
-  it('should not be shown if there is a 500 error with the `GET service_history` endpoint and the user is not a vet', async () => {
-    server.use(...mocks.getServiceHistory500);
+  context('when veteranStatus is null', () => {
```
Comment (erikphansen) [inline src/applications/personalization/profile/tests/components/alerts/Profile.data-unavailable-error.unit.spec.js:143]: These tests for when the `veteranStatus` is `null` already existed, I just refactored things a bit

File: src/applications/personalization/profile/tests/components/alerts/Profile.data-unavailable-error.unit.spec.js [hunk 3]
```diff
@@ -144,16 +140,40 @@ describe('Profile "Not all data available" error', () => {
...
+      initialState.user.profile.veteranStatus = null;
+      server.use(...mocks.getServiceHistory401);
+
+      await errorIsNotShownOnAnyPage(initialState);
+    });
   });
 
-  it('should not be shown if there is a 401 error with the `GET service_history` endpoint and the user is not a vet', async () => {
-    server.use(...mocks.getServiceHistory401);
+  context('when veteranStatus.status is "NOT_AUTHORIZED"', () => {
```
Comment (erikphansen) [inline src/applications/personalization/profile/tests/components/alerts/Profile.data-unavailable-error.unit.spec.js:161]: These tests are new. I confirmed they failed before I actually fixed the code

File: src/applications/personalization/profile/tests/e2e/nametag.cypress.spec.js [hunk 4]
```diff
@@ -52,6 +53,19 @@ describe('Profile NameTag', () => {
       nameTagRendersWithoutDisabilityRating();
     });
   });
+  context('when there is a 403 fetching the disability rating', () => {
```
Comment (erikphansen) [inline src/applications/personalization/profile/tests/e2e/nametag.cypress.spec.js:56]: This is unrelated but I wanted to add the test mainly for documentation purposes.

**Review by micahchiang — APPROVED** [review]

**Pull request merged**